File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        return ([""] + sorted([num[i - 2] + num[i - 1] + num[i] for i in range(2, len(num)) if num[i - 2] == num[i - 1] == num[i]]))[-1]

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        
        string ans;
        for (int i = 2; i < num.length(); i++){
            if ((num[i - 2] == num[i - 1]) && (num[i - 1] == num[i])){
                if (ans.length() == 0 or ans[0] < num[i]){
                    ans = num.substr(i - 2, 3);
                }
            } 
        }
        
        return ans;
    }
};